astropy/io/ascii/ui.py

```diff
@@ -641,6 +645,33 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader = get_reader(**guess_kwargs)
 
             reader.guessing = True
+
+            # Start off by checking what line endings are being used in file
```

Comment:
This should be moved outside the loop so that one doesn't find the short piece for every table type.